### ISSUE TITLE
Bug: [DeepRequired] Type errors when using `Array` interface extension

### ISSUE THREAD

**Issue by mtlewis** [ts-essentials/ts-essentials#366]
### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript`                           | `5.0.3` |
| `ts-essentials`                        | `9.3.1` |

### Repro Code

See TS playground, which includes `DeepRequired` from ts-essentials 9.3.1. Note the potentially naive fix suggested in the copied ts-essentials code, as well as the workaround at the bottom.

### TypeScript Playground

[here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgIQK7ADY2AOzgXzgDMoIQ4ByGAZwFoBTa6+nbAQw2ooG4AoXgPQC4AAwAi9emABK9AI7oo9ACYjipcgDcAnADoAzLoCMvegA9IsOMvoBjDGyVwYATzD04EqbIXAlygB4AFQA+OABeOCC4cxgWZWo4AFEoUiheODgAfjgfRRVgkIy4AC4omLM4nASUdCxcYpyg4qE4AAUIKvYMYmAzMtQcWzIQFngYAHcIOAxcRjgAI3oMCAnnaaVqCAxND3pU6BbhMujY+MSAQVS2FwDBgGscVZwizNam4pOKqpqAWTYwAFcER9nAANIAGjgwNBADVXtk4P9AV4ZPJ8oEwSEoai8n4CvCEV8ztVErI2MoIDgMC5kUCcCCoOCoTCmfDGrl6BSqTS6bj0fjMdjPJI0b5-AFCZ9yiSagB1Ln3Oms5nQhlwhE5BVsJUAgL88UFLE40V4iVSzLEyrnOAAZXoMHpjLgAFVNXaHfrTQKJW6iTLraTOdzqS57Y6VW6OeTKaHw17vD6Cn7padA-LFfHI+7tfd4waMQEU5aAz9Em0NMBmE7QVHMjkK2Qq-QE2LC8XSqWbQh8ByEMVMpkANpgtVwe70FwQIhRAC6tCyZQLguCI9nIT4g4I0rNBVCfF4uDiUCIbFsHlhHFQ9CuUBu3xtt5uAWoMCguAA5mEe-xhjhX3AABiEAQEu3qGoESALGwABei5wJeGDXk+LgEGEkT9lu0EwWUQ4UAoqBmBQs4DnArRytA9yOBAgzKCUpGtNhuH4aghHEXAbCJMu5pXje1wuNivD4NwQA)

### Expected Result

It should be possible to use `DeepRequired` with objects containing properties whose value is of a type which extends `Array<T>`. In situations like this, the property should be required to be set, but the expected members of the array should not change.

### Actual Result

In TS ~4, this works as expected. However, in TS5.0.3, there's a compilation error - it looks to be down to the fact that the DeepRequired type is modifying the members of the Array interface:

```
The types of '[Symbol.unscopables].length' are incompatible between these types.
    Type 'boolean | undefined' is not assignable to type 'boolean'.
      Type 'undefined' is not assignable to type 'boolean'.(2322)
```

**Comment by Beraliv** [comment]
The PR changes https://github.com/ts-essentials/ts-essentials/pull/367 have been merged and the will be included in major release that will be deployed by the end of April

**Comment by Beraliv** [comment]
It's released just now as a part of [v10.0.0](https://github.com/ts-essentials/ts-essentials/releases/tag/v10.0.0)
